lib/teleterm/clusters/storage.go

```diff
@@ -202,10 +204,15 @@ func (s *Storage) fromProfile(clusterName string) (*Cluster, error) {
 		return nil, trace.Wrap(err)
 	}
 
-	clusterURI := uri.NewClusterURI(clusterName)
+	clusterURI := uri.NewClusterURI(profileName)
+	if leafClusterName != "" {
+		clusterURI = clusterURI.AppendLeafCluster(leafClusterName)
+	}
 	return &Cluster{
-		URI:           clusterURI,
-		Name:          clusterName,
+		URI: clusterURI,
+		// TODO(ravicious): This should probably use leafClusterName if available, but at this point I'm
+		// worried that changing it might break something else in the app.
+		Name:          profileName,
```

Comment:
At the moment it doesn't cause any issues because Teleport Connect uses [another code path](https://github.com/gravitational/teleport/blob/d9deaff3d4465bd7654a0d698475d90b2b2d87fe/lib/teleterm/clusters/cluster_leaves.go#L41) to get names of leaf clusters.

Methods on `Storage` are used mostly to get a `Cluster` instance that lets us perform operations on the root/leaf cluster through `TeleportClient`.